readme.txt
--- DISCLAIMER ---
    FrogComposband may ruin your social life, work productivity, or
    daily exercise routine. You play the game at your own risk; in no
    event shall the FrogComposband authors owe you a new keyboard, or
    be liable to you for any other direct, indirect, punitive, magical
    or other injuries or damages of any nature whatsoever.

----- How to install -----

--- LINUX (instructions by Chris Kousky; lightly edited)
  Download and unpack the source archive, or clone the git repository:
    $ git clone https://github.com/sulkasormi/frogcomposband.git

  Make sure you have the appropriate development packages installed.
  For example, you might run (Ubuntu or Mint):
    $ sudo apt-get install autoconf gcc libc6-dev libncursesw5-dev libx11-dev

  From the root of the source archive:
    $ sh autogen.sh
    $ ./configure
    $ make clean
    $ make

  To install, you may need to elevate your credentials:
    $ su
    $ make install
    $ exit

  Then run frogcomposband as desired:
    $ frogcomposband -- -n<number of windows>  ## for normal ASCII graphics (recommended)
  or
    $ frogcomposband -g -- -n<# of windows>    ## for 8x8 tile graphics 

  If make install fails, or the game does not run properly after make install,
  try bypassing make install and just running the game from root.

  You can change game windows' font, location, and size, by environment 
  variables, for example:
    $ set env ANGBAND_X11_FONT '-*-*-medium-r-normal--24-*-*-*-*-*-iso8859-1'
    $ frogcomposband -- -n

  Then font size will be changed.

  You can set ANGBAND_X11_FONT_n for a specific window which has window number n.
  Window locations are ANGBAND_X11_AT_X_n and ANGBAND_X11_AT_Y_n.
  Window sizes are ANGBAND_X11_COLS_n and ANGBAND_X11_ROWS_n.

  (Thanks to Nick McConnell for implementing and improving building under Linux/MAC!)

  (ALT: For development, doing an install is undesirable. Try the following instead:
    $ sudo apt-get install clang-3.5 llvm-3.5
    $ sh autogen.sh
    $ ./configure SANITIZE_FLAGS=-fsanitize=address --with-no-install CC=clang-3.5
    $ make clean
    $ make -j4
    $ cp src/frogcomposband .
    $ ASAN_SYMBOLIZER_PATH=/usr/bin/llvm-symbolizer-3.5 ./frogcomposband -g -u<Savefile> -- -n1
    
    Note: Never pass sanitizer flags to CFLAGS as sanitizing configure causes it to fail.
    Instead, configure will append SANITIZE_FLAGS to CFLAGS and LDFLAGS once it has finished
    generating test programs. (see configure.ac)

    Note, you need to remember to copy the executable up after every build (TODO: Fix Makefile to do this!)
    Also, I like to develop with the address sanitizer enabled at all times. This works best with clang as
    your compiler, rather than gcc. Of course, 3.5 should be replaced with latest version of clang.).

    ***Fonts on Linux: My experience with linux has been that the fonts are just plain awful. Here, I document
    how I managed to install better fonts and use them in FrogComposband. I spent nearly a day on this, so I'm
    hoping this might prove useful to somebody else. But mostly, it is here to remind me how I did so I don't
    stumble so much next time.

    [1] Install some better fonts. For example
    $ sudo apt-get install fonts-liberation

    [2] See what fonts are on your system that FrogComposband can use:
    $ xlsfonts
    Notice that the new fonts aren't there!

    [3] Find where the new fonts were installed. For example
    $ fc-list | grep liberation
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-Italic.ttf: Liberation Sans Narrow:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf: Liberation Sans:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationMono-BoldItalic.ttf: Liberation Mono:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationSerif-Italic.ttf: Liberation Serif:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationMono-Bold.ttf: Liberation Mono:style=Bold
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-Regular.ttf: Liberation Sans Narrow:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationSerif-Bold.ttf: Liberation Serif:style=Bold
    /usr/share/fonts/truetype/liberation/LiberationMono-Regular.ttf: Liberation Mono:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationSans-Italic.ttf: Liberation Sans:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationSerif-BoldItalic.ttf: Liberation Serif:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-BoldItalic.ttf: Liberation Sans Narrow:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationMono-Italic.ttf: Liberation Mono:style=Italic
    /usr/share/fonts/truetype/liberation/LiberationSans-BoldItalic.ttf: Liberation Sans:style=Bold Italic
    /usr/share/fonts/truetype/liberation/LiberationSerif-Regular.ttf: Liberation Serif:style=Regular
    /usr/share/fonts/truetype/liberation/LiberationSansNarrow-Bold.ttf: Liberation Sans Narrow:style=Bold
    /usr/share/fonts/truetype/liberation/LiberationSans-Bold.ttf: Liberation Sans:style=Bold

    [4] Bang head on wall in frustration!!

    [5] Tell X Server about the fonts:
    $ cd /usr/share/fonts/truetype/liberation/
    $ su
    $ mkfontscale
    $ mkfontdir
    $ exit
    $ cat fonts.dir
    ...
    LiberationMono-Regular.ttf -misc-liberation mono-medium-r-normal--0-0-0-0-m-0-adobe-standard
    LiberationMono-Regular.ttf -misc-liberation mono-medium-r-normal--0-0-0-0-m-0-ascii-0
    LiberationMono-Regular.ttf -misc-liberation mono-medium-r-normal--0-0-0-0-m-0-ibm-cp437
    ... [and many more]

    $ xset +fp /usr/share/fonts/truetype/liberation
    $ xset fp rehash
    $ xlsfonts | grep liberation
    ...
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-1
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-10
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-13
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-15
    -misc-liberation sans-medium-i-normal--0-0-0-0-p-0-iso8859-16
    ... [and many more]

    [6] Now, try to find a font you like. For example:
    $ cd [path to frogcomposband]
    ~/Src/frogcomposband> ANGBAND_X11_FONT='-misc-liberation mono-medium-r-normal--20-0-0-0-m-0-iso8859-1' ./frogcomposband -mx11
    You can play with the point size since these are vector fonts. I chose 20pt since my eyes suck!

    [7] Smile. Grab beer to recover from step [4].

--- Curses

  Curses is for Linux, of course, so everything said above also applies here.
  To run with a single 'big' terminal, simply run, for example:
    $ ./frogcomposband -mgcu -uCrusher

  To add additional terminal windows, you need to specify sub-options. You can
  configure, from the command line, a strip of terminals on the right hand side
  of the screen, or on the bottom of the screen, or both.

  For example:
    $ ./frogcomposband -mgcu -uCrusher -- -right 57x26,*

  This specifies that the right hand strip will be be 57 columns wide, and will
  contain 2 additional terminals. The first one, on top, will be 57x26 (i.e., 26
  rows high) while the second terminal will be 57x(LINES-26) (i.e., * means fill
  to whatever is leftover). These terminals will be numbered 1 and 2, respectively.

  Another example:
    $ ./frogcomposband -mgcu -uCrusher -- -bottom -bottom *x10

  This adds a bottom strip 10 rows high, and this strip will contain a single
  additional terminal window (numbered as Term-1) that will be as wide as the
  screen allows (i.e, Term-1 will be COLSx10).

  Finally, you can combine the -right and -bottom commands, in either order.
  For example:
    $ ./frogcomposband -mgcu -uCrusher -- -right 57x26,* -bottom *x10

  Here, Term-1 and Term-2 are on the right strip, sized as 57x26 and 57x(LINES-26),
  while Term-3 is on the bottom strip, sized (COLS-57)x10. Term-0, the Main Terminal,
  is, of course, located in the top left and sized as (COLS-57)x(LINES-10). The
  map terminal will always use as much space as possible.

  Finally, the order of the -right and -bottom command is significant, as it affects
  both the child terminal numbering and also resolves the conflict over the overlap
  region in the bottom-right corner of the screen.

  For example, the meaning of the following should now be clear:
    $ ./frogcomposband -mgcu -uCrusher -- -bottom *x10 -right 57x26,*

  You cannot specify more than 7 child terminals.

--- Windows

  Download the binary archive for Windows.  Unzip it to any location that you 
  will have full permissions and launch frogcomposband to play.

  To compile the source code in MinGW:
  $ ./autogen.sh
  $ ./configure --enable-win
  $ make

  MinGW sometimes randomly chokes on one of these steps, in which case you
  will need to redo that step.


----------  Basics  ------------

  The in-game documentation has been updated for FrogComposband 7.1.liquorice.
  Press '?' to activate the help system, then select your favorite topic; e.g.
  press 'a' twice for General Information; press 'a' followed by 'b' for an
  introduction to the help files and how to best read them [yes, this is that
  kind of manual!]; press 'a' followed by 'j' for the Newbie Guide. If you
  get lost in the depths of the help system, press '!' to return to the
  help's main page.

  Helpfiles are also available as HTML files for viewing outside the game
  in your favorite browser; for example, the General Information page is at
  lib/help/html/general.html and the Newbie Guide at lib/help/html/tang.html.

  If you are still stuck, you can usually find friendly people at angband.live
  who will try to help you out.

---------  Commands  ----------- 

  Again, please refer to the in-game help; press '?' for help and then 'c'
  for Commands. Choose the section that interests you ("Command Descriptions"
  is a separate, lengthy file that explains all commands in detail).

  Or open lib/help/html/command.html or lib/help/html/commdesc.html in your browser.

--- Most of this file inherited from PosChengband, updated for FrogComposband 7.1.liquorice.


src/changes
Changes in FrogComposband 7.1.salmiak:

* Added new dungeons and randomized which dungeons actually appear

* Added new objects, ego types, artifacts, item curses, terrain features and
random artifact names

* Added new monsters (including the reintroduction of the Greater Cyber
Wyrm Angel Daemon Lich as an exclusively postgame monster)

* Added new player-monster race (Mummy) based on Maledicts from Pos-R

* Moved to a 32-color palette and changed the colors of many monsters
-- NOTES: If your platform does not support more than 16 colors, or
   FrogComposband's coloring code for your platform is faulty, compile with
   --disable-xtra-colours to retain use of 16 colors only. If your platform
   does support more than 16 colors but you wish to stick to 16 (from
   nostalgic reasons or for greater contrast between colors), load
   user-lim.prf (typing &4 in-game does this automatically) or otherwise
   modify the colors with the '&' command
-- Special thanks to Oneiron for helping me debug the 32-color palette

* Town services could enchant multiple weapons for the price of one; this has
been fixed (thanks to Thraalbee for reporting)

* Directions were given for Heaven and Hell even if the player had not won;
this has been fixed (thanks to Marty and ster for reporting)

* Fixed bug when Teleport Level was used after completing Crystal Quest
(thanks to bostock for reporting)

* Fixed bug when stairs were taken just when recall kicked in
(thanks to wobbly and bostock for reporting)

* Fixed bug with hand grenade explosions sometimes causing a "player"
fake monster to temporarily appear (thanks to wobbly for reporting)

* Fixed bug with multi-effect monster blows sometimes killing an opponent
repeatedly (thanks to bostock for reporting)

* Fixed bug with autotargeting the mount for hasting and healing
(thanks to bostock for reporting)

* Fixed bug with devices in wilderness rooms generating with no effect
(thanks to bostock for reporting)

* Fixed bug with standard artifacts disappearing from the floor if they had
been carried there from another level (thanks to bostock for reporting)

* Fixed bug with Ninjas continuing to attack if Kawarimi was triggered
mid-turn (thanks to bostock for reporting)

* Timed monster statuses were not processed in the overworld; this has been
fixed (thanks to bostock for reporting)

* Fixed old bug with statue stacking (thanks to kt for reporting)

* Fixed infinite loop on junk autoget when always_pickup was on and
the Easy Destroyer was not on (thanks to DavidMedley for reporting)

* Fixed crash when the object list subwindow was on and an item with
specific properties was equipped from the floor (thanks to caruso for
reporting)

* A potion of Restore Mana thrown from the floor could cause a phantom copy
of itself to explode; this has been fixed

* Bumping into mountains while confused consumed no time; this has been fixed
(thanks to Hugo for reporting)

* Fixed weirdness on player beholders with mutational innate attacks
(thanks to Karkaroth for reporting)

* Fixed annoying and inconsistent handling of erratic movement on mounts
(thanks to budswell for reporting)

* Fixed bug with newly acquired locked mutations not affecting regeneration
properly (thanks to ster for reporting)

* Fixed bug with insurance and player-elemental item destruction

* Fixed bug with inappropriate handling of insured items left behind on
a level the player subsequently returned to

* Fixed crash on shop shuffle if too many items had been reserved

* Fixed ancient bugs with the wilderness terrain-selection algorithm, and
adjusted some numbers that had been optimized for the buggy version

* Fixed various types of weirdness with Bunshin; it is now generally less
useful against auras, but more useful in a number of other situations where
it previously did not apply

* Fixed minor bug with innate attack damage calculations

* Fixed weirdness with charmed clones

* Fixed yet another loophole to anti-scumming item/gold drop limits

* Fixed incorrect pluralization of some quiver egos

* Fixed weirdness with door placement in cavern areas

* Fixed occasional weirdness with stair placement after quest completion

* Multiple items could be reserved for the price of one; this has been fixed

* Turncounts in character dumps now properly account for midnight starts
(this does not apply retroactively)

* Changed the power of confusion effects, they should now at least resemble
the claimed powers

* Fixed issue with the powers of Slow Monster bolt spells
(thanks to Saru for reporting)

* Allowed Samurai to view damage averages for Kendo attack techniques
(thanks to ster for suggesting)

* Renamed weak Beam of Light to "Moonbeam" to more clearly distinguish it
from strong Beam of Light (thanks to bostock for suggesting)

* Slightly relaxed capture-balling requirements for hostile monsters to
prevent some low-HP monsters from being captureable at all
(thanks to caruso for bringing this up)

* Added option (on by default) to limit quantity prompts in shops to the
highest amount that is both available and affordable
(thanks to caruso for suggesting)

* Added option (off by default) to indicate unknown item flavors in shops
(thanks to caruso for suggesting)

* Added option (off by default) to display the nominal energy cost of the
most recent action (thanks to Bucephalus for suggesting)

* Added option (on by default) to extend the unknown attribute indicator '?'
to cover random dragon resistances (thanks to bostock for bringing this up)

* Combined the empty_levels option with the ironman_empty_levels option

* Allowed using Home/End and 7/1 to immediately move to the beginning or end
of a long home or museum inventory (thanks to Melum for suggesting)

* Renamed the "Cause Wounds" monster spells/auras to various types of "Curse",

src/mut.h
/* New Mutation System, carefully crafted to be backwards compatible with the old ... 
   You are free to add new mutations (though extending MUT_FLAG_SIZE requires save file
   work), and place activatable mutations where ever you please.
*/
#define MUT_SPIT_ACID      0   /* Start of old mutations */
#define MUT_BR_FIRE        1
#define MUT_HYPN_GAZE      2
#define MUT_TELEKINESIS    3
#define MUT_TELEPORT       4
#define MUT_MIND_BLAST     5
#define MUT_RADIATION      6
#define MUT_VAMPIRISM      7
#define MUT_SMELL_METAL    8
#define MUT_SMELL_MONSTERS 9
#define MUT_BLINK          10
#define MUT_EAT_ROCK       11
#define MUT_SWAP_POS       12
#define MUT_SHRIEK         13
#define MUT_ILLUMINE       14
#define MUT_DET_CURSE      15
#define MUT_BERSERK        16
#define MUT_POLYMORPH      17
#define MUT_MIDAS_TOUCH    18
#define MUT_GROW_MOLD      19
#define MUT_RESIST         20
#define MUT_EARTHQUAKE     21
#define MUT_EAT_MAGIC      22
#define MUT_WEIGH_MAGIC    23
#define MUT_STERILITY      24
#define MUT_PANIC_HIT      25
#define MUT_DAZZLE         26
#define MUT_LASER_EYE      27
#define MUT_RECALL         28
#define MUT_BANISH         29
#define MUT_COLD_TOUCH     30
#define MUT_LAUNCHER       31

#define MUT_BERS_RAGE      32
#define MUT_COWARDICE      33
#define MUT_TELEPORT_RND   34
#define MUT_ALCOHOL        35
#define MUT_HALLUCINATION  36
#define MUT_FLATULENT      37
#define MUT_SCORPION_TAIL  38
#define MUT_HORNS          39
#define MUT_BEAK           40
#define MUT_ATTRACT_DEMON  41
#define MUT_PROD_MANA      42
#define MUT_SPEED_FLUX     43
#define MUT_BANISH_ALL_RND 44
#define MUT_EAT_LIGHT      45
#define MUT_TRUNK          46
#define MUT_ATTRACT_ANIMAL 47
#define MUT_TENTACLES      48
#define MUT_RAW_CHAOS      49
#define MUT_NORMALITY      50
#define MUT_WRAITH         51
#define MUT_POLY_WOUND     52
#define MUT_WASTING        53
#define MUT_ATTRACT_DRAGON 54
#define MUT_RANDOM_TELEPATHY 55
#define MUT_NAUSEA         56
#define MUT_CHAOS_GIFT     57
#define MUT_SHADOW_WALK    58
#define MUT_WARNING        59
#define MUT_INVULN         60
#define MUT_SP_TO_HP       61
#define MUT_HP_TO_SP       62
#define MUT_FUMBLING       63

#define MUT_HYPER_STR      64
#define MUT_PUNY           65
#define MUT_HYPER_INT      66
#define MUT_MORONIC        67
#define MUT_RESILIENT      68
#define MUT_XTRA_FAT       69
#define MUT_ALBINO         70
#define MUT_FLESH_ROT      71
#define MUT_SILLY_VOICE    72
#define MUT_BLANK_FACE     73
#define MUT_ILL_NORM       74
#define MUT_XTRA_EYES      75
#define MUT_MAGIC_RES      76
#define MUT_XTRA_NOISE     77
#define MUT_INFRAVISION    78
#define MUT_XTRA_LEGS      79
#define MUT_SHORT_LEG      80
#define MUT_ELEC_AURA      81
#define MUT_FIRE_AURA      82
#define MUT_WARTS          83
#define MUT_SCALES         84
#define MUT_STEEL_SKIN     85
#define MUT_WINGS          86
#define MUT_FEARLESS       87
#define MUT_REGEN          88
#define MUT_ESP            89
#define MUT_LIMBER         90
#define MUT_ARTHRITIS      91
#define MUT_BAD_LUCK       92
#define MUT_VULN_ELEM      93
#define MUT_MOTION         94
#define MUT_GOOD_LUCK      95   /* End of the old mutations */

#define MUT_FAST_LEARNER   96
#define MUT_WEAPON_SKILLS  97
#define MUT_SUBTLE_CASTING 98
#define MUT_PEERLESS_SNIPER 99
#define MUT_UNYIELDING     100
#define MUT_AMBIDEXTROUS   101
#define MUT_UNTOUCHABLE    102
#define MUT_LOREMASTER     103
#define MUT_ARCANE_MASTERY 104
#define MUT_EVASION        105
#define MUT_POTION_CHUGGER 106
#define MUT_ONE_WITH_MAGIC 107
#define MUT_PEERLESS_TRACKER 108
#define MUT_INFERNAL_DEAL 109
#define MUT_MERCHANTS_FRIEND 110
#define MUT_SACRED_VITALITY 111
#define MUT_CULT_OF_PERSONALITY 112
#define MUT_FLEET_OF_FOOT 113
#define MUT_ASTRAL_GUIDE 114
#define MUT_DEMONIC_GRASP 115
#define MUT_WEIRD_MIND 116
#define MUT_FANTASTIC_FRENZY 117
#define MUT_BLACK_MARKETEER 118
#define MUT_SPEED_READER 119
#define MUT_FELL_SORCERY 120
#define MUT_TREAD_SOFTLY 121

#define MUT_DRACONIAN_SHIELD 122
#define MUT_DRACONIAN_MAGIC_RES 123
#define MUT_DRACONIAN_STRIKE 124
#define MUT_DRACONIAN_BREATH 125
#define MUT_DRACONIAN_REGEN 126
#define MUT_DRACONIAN_KIN 127
#define MUT_DRACONIAN_LORE 128
#define MUT_DRACONIAN_RESISTANCE 129
#define MUT_DRACONIAN_METAMORPHOSIS 130

/* Frog(spawn) mutations */
#define MUT_EASY_TIRING       131
#define MUT_SENSITIVE_EYES    132
#define MUT_NO_INHIBITIONS    133
#define MUT_WAYBREAD_INTO     134
#define MUT_PULTITIS          135
#define MUT_HYPOCHONDRIA      136
#define MUT_PURPLE_GIFT       137
#define MUT_INSPIRED_SMITHING 138
#define MUT_IMPOTENCE         139
#define MUT_STRONG_MIND       140

/* Vortex mutations */
#define MUT_VORTEX_MELEE      141
#define MUT_VORTEX_SPEED      142
#define MUT_VORTEX_CONTROL    143

/* More frog muts */
#define MUT_EASY_TIRING2      144
#define MUT_LIMP              145

/* Human weaknesses */
#define MUT_HUMAN_STR         146
#define MUT_HUMAN_INT         147
#define MUT_HUMAN_WIS         148
#define MUT_HUMAN_DEX         149
#define MUT_HUMAN_CON         150
#define MUT_HUMAN_CHR         151

#define MAX_MUTATIONS      152   /* see also MUT_FLAG_SIZE in defines.h */

typedef enum {
    MUT_RATING_AWFUL = -2,
    MUT_RATING_BAD   = -1,
    MUT_RATING_AVERAGE = 0,
    MUT_RATING_GOOD = 1,
    MUT_RATING_GREAT = 2
} mutation_rating;

typedef enum {
    MUT_TYPE_ACTIVATION = 0x01,    /* Mutation activates as a racial power (e.g. Harden to Elements) */
    MUT_TYPE_EFFECT = 0x02,     /* Mutation has a passive effect (e.g. Flatulence)*/
    MUT_TYPE_BONUS = 0x04,      /* Mutation confers a passive bonus (e.g. +4 Str) */
} mutation_type;

typedef bool (*mut_pred)(int mut_idx);
extern bool mut_good_pred(int mut_idx);
extern bool mut_bad_pred(int mut_idx);
extern bool mut_human_pred(int mut_idx);
extern bool mut_demigod_pred(int mut_idx);
extern bool mut_draconian_pred(int mut_idx);
extern bool mut_vortex_pred(int mut_idx);
extern bool mut_unlocked_pred(int mut_idx);

extern void mut_calc_bonuses(void);
extern void mut_get_flags(u32b flgs[OF_ARRAY_SIZE]);
extern void mut_calc_stats(s16b stats[MAX_STATS]);
extern int  mut_gain_random_aux(mut_pred pred);
extern int  mut_count(mut_pred pred);
extern void mut_do_cmd_knowledge(void);
extern void mut_display(doc_ptr doc);
extern bool mut_gain(int mut_idx);
extern int mut_gain_choice(mut_pred pred);
extern bool mut_gain_random(mut_pred pred);
extern int  mut_get_powers(power_info* spells, int max);
extern void mut_lock(int mut_idx);
extern bool mut_locked(int mut_idx);
extern bool mut_lose(int mut_idx);
extern void mut_lose_all(void);
extern bool mut_lose_random(mut_pred pred);
extern void mut_name(int i, char* buf);
extern void mut_help_desc(int i, char* buf);
extern bool mut_present(int mut_idx);
extern void mut_process(void);
extern int  mut_rating(int mut_idx);
extern int  mut_regenerate_mod(void);
extern int  mut_type(int mut_idx);
extern void mut_unlock(int mut_idx);

/* Non-spell mutations ... */
extern void albino_mut(int cmd, variant *res);
extern void alcohol_mut(int cmd, variant *res);
extern void ambidexterity_mut(int cmd, variant *res);
extern void arcane_mastery_mut(int cmd, variant *res);
extern void arthritis_mut(int cmd, variant *res);
extern void astral_guide_mut(int cmd, variant *res);
extern void attract_animal_mut(int cmd, variant *res);
extern void attract_demon_mut(int cmd, variant *res);
extern void attract_dragon_mut(int cmd, variant *res);
extern void bad_luck_mut(int cmd, variant *res);
extern void black_marketeer_mut(int cmd, variant *res);
extern void beak_mut(int cmd, variant *res);
extern void berserk_rage_mut(int cmd, variant *res);
extern void blank_face_mut(int cmd, variant *res);
extern void cerebral_pultitis_mut(int cmd, variant *res);
extern void chaos_deity_mut(int cmd, variant *res);
extern void cowardice_mut(int cmd, variant *res);
extern void cult_of_personality_mut(int cmd, variant *res);
extern void eat_light_mut(int cmd, variant *res);
extern void demonic_grasp_mut(int cmd, variant *res);

extern void draconian_breath_mut(int cmd, variant *res);
extern void draconian_kin_mut(int cmd, variant *res);
extern void draconian_lore_mut(int cmd, variant *res);
extern void draconian_magic_resistance_mut(int cmd, variant *res);
extern void draconian_metamorphosis_mut(int cmd, variant *res);
extern void draconian_regen_mut(int cmd, variant *res);
extern void draconian_resistance_mut(int cmd, variant *res);
extern void draconian_shield_mut(int cmd, variant *res);
extern void draconian_strike_mut(int cmd, variant *res);

extern void easy_tiring_mut(int cmd, variant *res);
extern void easy_tiring_II_mut(int cmd, variant *res);
extern void einstein_mut(int cmd, variant *res);
extern void elec_aura_mut(int cmd, variant *res);
extern void evasion_mut(int cmd, variant *res);
extern void extra_eyes_mut(int cmd, variant *res);
extern void extra_legs_mut(int cmd, variant *res);
extern void extra_noise_mut(int cmd, variant *res);
extern void fantastic_frenzy_mut(int cmd, variant *res);
extern void fast_learner_mut(int cmd, variant *res);
extern void fat_mut(int cmd, variant *res);
extern void fearless_mut(int cmd, variant *res);
extern void fell_sorcery_mut(int cmd, variant *res);
extern void fire_aura_mut(int cmd, variant *res);
extern void flatulence_mut(int cmd, variant *res);
extern void fleet_of_foot_mut(int cmd, variant *res);
extern void fumbling_mut(int cmd, variant *res);
extern void good_luck_mut(int cmd, variant *res);
extern void hallucination_mut(int cmd, variant *res);
extern void he_man_mut(int cmd, variant *res);
extern void horns_mut(int cmd, variant *res);
extern void hypochondria_mut(int cmd, variant *res);
extern void illusion_normal_mut(int cmd, variant *res);
extern void impotence_mut(int cmd, variant *res);
extern void infernal_deal_mut(int cmd, variant *res);
extern void infravision_mut(int cmd, variant *res);
extern void inspired_smithing_mut(int cmd, variant *res);
extern void invulnerability_mut(int cmd, variant *res);
extern void limber_mut(int cmd, variant *res);
extern void limp_mut(int cmd, variant *res);
extern void loremaster_mut(int cmd, variant *res);
extern void magic_resistance_mut(int cmd, variant *res);
extern void merchants_friend_mut(int cmd, variant *res);
extern void moron_mut(int cmd, variant *res);

extern void human_chr_mut(int cmd, variant *res);
extern void human_con_mut(int cmd, variant *res);
extern void human_dex_mut(int cmd, variant *res);
extern void human_int_mut(int cmd, variant *res);
extern void human_str_mut(int cmd, variant *res);
extern void human_wis_mut(int cmd, variant *res);

extern void motion_mut(int cmd, variant *res);
extern void nausea_mut(int cmd, variant *res);
extern void no_inhibitions_mut(int cmd, variant *res);
extern void normality_mut(int cmd, variant *res);
extern void one_with_magic_mut(int cmd, variant *res);
extern void peerless_sniper_mut(int cmd, variant *res);
extern void peerless_tracker_mut(int cmd, variant *res);
extern void polymorph_wounds_mut(int cmd, variant *res);
extern void potion_chugger_mut(int cmd, variant *res);
extern void produce_mana_mut(int cmd, variant *res);
extern void puny_mut(int cmd, variant *res);
extern void purple_mut(int cmd, variant *res);
extern void random_banish_mut(int cmd, variant *res);
extern void random_teleport_mut(int cmd, variant *res);
extern void random_telepathy_mut(int cmd, variant *res);
extern void raw_chaos_mut(int cmd, variant *res);
extern void regeneration_mut(int cmd, variant *res);
extern void resilient_mut(int cmd, variant *res);
extern void rotting_flesh_mut(int cmd, variant *res);
extern void sacred_vitality_mut(int cmd, variant *res);
extern void scales_mut(int cmd, variant *res);
extern void scorpion_tail_mut(int cmd, variant *res);
extern void sensitive_eyes_mut(int cmd, variant *res);
extern void shadow_walk_mut(int cmd, variant *res);
extern void short_legs_mut(int cmd, variant *res);
extern void silly_voice_mut(int cmd, variant *res);
extern void speed_flux_mut(int cmd, variant *res);
extern void speed_reader_mut(int cmd, variant *res);
extern void steel_skin_mut(int cmd, variant *res);
extern void strong_mind_mut(int cmd, variant *res);
extern void subtle_casting_mut(int cmd, variant *res);
extern void telepathy_mut(int cmd, variant *res);
extern void tentacles_mut(int cmd, variant *res);
extern void tread_softly_mut(int cmd, variant *res);
extern void trunk_mut(int cmd, variant *res);
extern void untouchable_mut(int cmd, variant *res);
extern void unyielding_mut(int cmd, variant *res);

extern void vortex_melee_mut(int cmd, variant *res);
extern void vortex_speed_mut(int cmd, variant *res);
extern void vortex_control_mut(int cmd, variant *res);

extern void vulnerability_mut(int cmd, variant *res);
extern void warning_mut(int cmd, variant *res);
extern void warts_mut(int cmd, variant *res);
extern void wasting_mut(int cmd, variant *res);
extern void waybread_into_mut(int cmd, variant *res);
extern void weapon_skills_mut(int cmd, variant *res);
extern void weird_mind_mut(int cmd, variant *res);
extern void wings_mut(int cmd, variant *res);
extern void wraith_mut(int cmd, variant *res);

